pkgs/servers/etcd/3.5.nix

```
@@ -0,0 +1,76 @@
+{ lib, buildGoModule, fetchFromGitHub, symlinkJoin }:
+
+let
+  etcdVersion = "3.5.1";
+  etcdSrc = fetchFromGitHub {
+    owner = "etcd-io";
+    repo = "etcd";
+    rev = "v${etcdVersion}";
+    sha256 = "sha256-Ip7JAWbZBZcc8MXd+Sw05QmTs448fQXpQ5XXo6RW+Gs=";
+  };
+
+  commonMeta = with lib; {
+    description = "Distributed reliable key-value store for the most critical data of a distributed system";
+    license = licenses.asl20;
+    homepage = "https://etcd.io/";
+    maintainers = with maintainers; [ offline zowoq endocrimes ];
+    platforms = platforms.unix;
+  };
+
+  etcdserver = buildGoModule rec {
+    pname = "etcdserver";
+    version = etcdVersion;
+
+    vendorSha256 = "sha256-hJzmxCcwN6MTgE0NpjtFlm8pjZ83clQXv1k5YM8Gmes=";
+
+    doCheck = true;
+
+    src = etcdSrc;
+    modRoot = "./server";
+
+    postInstall = ''
+      mv $out/bin/server $out/bin/etcd
+    '';
```

People will likely care about cross builds for this package so shouldn't use `postInstall` as it will break.

Untested but it should work:
```suggestion
    postBuild = ''
      mv go/bin/{server,etcd}
    '';
```

---

Updated 👍